code/datums/gamemode/role/ninja.dm

```diff
@@ -493,15 +493,19 @@ Helpers For Both Variants
 	daemon = new /datum/daemon/teleport(src,"Weakness",null)
 
 /obj/item/weapon/melee/energy/sword/ninja/toggleActive(mob/user, var/togglestate = "")
+	if(togglestate) //override
+		..()
+		checkdroppable()
+		return
 	if(isninja(user))
 		..()
+		checkdroppable()
 	else
 		to_chat(user,"<span class='warning'>There's no buttons on it.</span>")
 		return
-	if(active)
-		cant_drop = TRUE
-	else
-		cant_drop = FALSE
+
+/obj/item/weapon/melee/energy/sword/ninja/proc/checkdroppable()
+	return cant_drop = active //they should be the same value every time
```

Comment:
Shouldn't this be ==

Reply:
Nah, it's just setting the former to the latter. Maybe the proc isn't even necessary since it's only used twice.